icu4c/source/test/intltest/dtptngts.cpp

```diff
@@ -1631,4 +1632,129 @@ void IntlTestDateTimePatternGeneratorAPI::testBestPattern() {
+                        localeAndResultsPtr->localeID, u_errorName(status));
+        } else {
+            doDTPatternTest(dtpg.getAlias(), skeletons, localeAndResultsPtr);
+        }
+    }
+    // Test getting and modifying date-time combining patterns
+    status = U_ZERO_ERROR;
+    dtpg.adoptInstead(DateTimePatternGenerator::createInstance(Locale::getEnglish(), status));
+    if (U_FAILURE(status)) {
+        dataerrln("FAIL: DateTimePatternGenerator::createInstance #2 for locale en: %s", u_errorName(status));
+    } else {
+        // Test style out of range
+        status = U_ZERO_ERROR;
+        const UnicodeString& dtFormat0 = dtpg->getDateTimeFormat(UDAT_NONE, status);
+        int32_t dtFormat0Len = dtFormat0.length();
+        if (status != U_ILLEGAL_ARGUMENT_ERROR || dtFormat0Len != 0) {
+            errln("ERROR: getDateTimeFormat with invalid style, expected U_ILLEGAL_ARGUMENT_ERROR and lero-length string, "
+                  "got %s with length %d", u_errorName(status), dtFormat0Len);
+        }
+        // Test normal getting and setting
+        for (int32_t patStyle = 0; patStyle < kNumDateTimePatterns; patStyle++) {
+            char bExpect[64];
+            char bGet[64];
+            status = U_ZERO_ERROR;
+            const UnicodeString& dtFormat1 = dtpg->getDateTimeFormat((UDateFormatStyle)patStyle, status);
+            if (U_FAILURE(status)) {
+                errln("FAIL: getDateTimeFormat for en before mod, style %d, get %s", patStyle, u_errorName(status));
+            } else  if (dtFormat1 != enDTPatterns[patStyle]) {
+                enDTPatterns[patStyle].extract(0, enDTPatterns[patStyle].length(), bExpect, 64);
+                dtFormat1.extract(0, dtFormat1.length(), bGet, 64);
+                errln("ERROR: getDateTimeFormat for en before mod, style %d, expect \"%s\", get \"%s\"",
+                        patStyle, bExpect, bGet);
+            }
+            status = U_ZERO_ERROR;
+            dtpg->setDateTimeFormat((UDateFormatStyle)patStyle, modDTPatterns[patStyle], status);
+            if (U_FAILURE(status)) {
+                errln("FAIL: setDateTimeFormat for en, style %d, get %s", patStyle, u_errorName(status));
+            } else {
+                const UnicodeString& dtFormat2 = dtpg->getDateTimeFormat((UDateFormatStyle)patStyle, status);
+                if (U_FAILURE(status)) {
+                    errln("FAIL: getDateTimeFormat for en after  mod, style %d, get %s", patStyle, u_errorName(status));
+                } else if (dtFormat2 != modDTPatterns[patStyle]) {
+                    modDTPatterns[patStyle].extract(0, modDTPatterns[patStyle].length(), bExpect, 64);
+                    dtFormat2.extract(0, dtFormat2.length(), bGet, 64);
+                    errln("ERROR: getDateTimeFormat for en after mod, style %d, expect \"%s\", get \"%s\"",
+                            patStyle, bExpect, bGet);
+                }
+            }
+        }
```

Comment:
Do you have a test for the new behavior for the old getter and setter?  Did I miss it?